homeassistant/components/climate/mysensors.py

```diff
@@ -68,42 +53,38 @@ def unit_of_measurement(self):
     @property
     def current_temperature(self):
         """Return the current temperature."""
-        if len(self._values) != 0:
-            return self._values[self.gateway.const.SetReq.V_TEMP]
-        else:
-            return None
+        return self._values.get(self.gateway.const.SetReq.V_TEMP)
 
     @property
     def target_temperature(self):
         """Return the temperature we try to reach."""
-        if len(self._values) != 0:
-            return self._values[self.gateway.const.SetReq.V_HVAC_SETPOINT_COOL]
-        else:
+        set_req = self.gateway.const.SetReq
+        if set_req.V_HVAC_SETPOINT_COOL in self._values and \
+                set_req.V_HVAC_SETPOINT_HEAT in self._values:
             return None
+        temp = self._values.get(set_req.V_HVAC_SETPOINT_COOL)
+        if temp is None:
+            temp = self._values.get(set_req.V_HVAC_SETPOINT_HEAT)
+        return temp
 
     @property
     def target_temperature_high(self):
         """Return the highbound target temperature we try to reach."""
-        if len(self._values) != 0:
-            return self._values[self.gateway.const.SetReq.V_HVAC_SETPOINT_COOL]
-        else:
-            return None
+        set_req = self.gateway.const.SetReq
+        if set_req.V_HVAC_SETPOINT_COOL in self._values:
+            return self._values.get(set_req.V_HVAC_SETPOINT_COOL)
 
     @property
     def target_temperature_low(self):
         """Return the lowbound target temperature we try to reach."""
-        if len(self._values) != 0:
-            return self._values[self.gateway.const.SetReq.V_HVAC_SETPOINT_COOL]
-        else:
-            return None
+        set_req = self.gateway.const.SetReq
+        if set_req.V_HVAC_SETPOINT_HEAT in self._values:
```

Comment:
Check that `V_HVAC_SETPOINT_COOL` is in `self._values`. We need both heat and cool setpoints if we should use high and low targets.